test/framework/src/main/java/org/elasticsearch/test/rest/yaml/section/DoSection.java

```diff
@@ -293,18 +293,21 @@ void checkWarningHeaders(final List<String> warningHeaders, final Version master
             if (matches) {
                 final String message = matcher.group(1);
                 // noinspection StatementWithEmptyBody
-                if (masterVersion.before(Version.V_7_0_0)
+                if ((masterVersion.before(Version.V_7_0_0)
                         && message.equals("the default number of shards will change from [5] to [1] in 7.0.0; "
                         + "if you wish to continue using the default of [5] shards, "
-                        + "you must manage this on the create index request or with an index template")) {
+                        + "you must manage this on the create index request or with an index template"))
+                    || message.startsWith("[types removal]")) {
                     /*
-                     * This warning header will come back in the vast majority of our tests that create an index when running against an
-                     * older master. Rather than rewrite our tests to assert this warning header, we assume that it is expected.
+                     * We ignore two classes of warning headers:
+                     * - The default number of shards warning will come back in the vast majority of our tests that
+                     * create an index when running against an older master. Rather than rewrite our tests to assert
+                     * this warning header, we assume that it is expected.
+                     * - We skip warnings related to types deprecation so that we can continue to run the many
+                     * mixed-version tests that used typed APIs.
                      */
-                } else {
-                    if (expected.remove(message) == false) {
-                        unexpected.add(header);
-                    }
+                } else if (expected.remove(message) == false) {
+                    unexpected.add(header);
```

Comment:
I wonder if this change requires some adapting on language clients test runners. I think that some clients don't support headers, hence they can't support the warning feature, so this does not affect them (they already skip all the tests around warnings and don't fail if warnings are returned). Not sure what other clients that support headers do though. If they do fail when a warning is returned, which they should, their runner should be adapted.